test/userController.test.js

```diff
@@ -50,6 +50,243 @@ afterAll(async () => {
+    expect(response.body).toHaveProperty('msg', 'Please provide all values');
+  });
+});
+
+describe('PATCH /api/v1/users/updateUserPassword endpoint', () => {
+  it("should update the current user's password", async () => {
+    // Arrange: Log in and get a signed cookie with valid test user credentials
+    const signedCookie = await loginAndReturnCookie(testUserCredentials);
+
+    // Arrange: Define the old and new passwords
+    const oldPassword = 'secret';
+    const newPassword = 'newSecret';
+
+    // Act: Attempt to update the user's password using the authenticated cookie
+    const response = await request(app)
+      .patch('/api/v1/users/updateUserPassword')
+      .set('Cookie', [signedCookie])
+      .send({ oldPassword, newPassword });
+
+    // Assert: Check the response status for success
+    expect(response.status).toBe(200);
+    expect(response.body).toHaveProperty('msg', 'Success! Password Updated.');
+  });
+
+  it('should return an error if old password is incorrect', async () => {
+    // Arrange: Log in and get a signed cookie with valid test user credentials
+    const signedCookie = await loginAndReturnCookie(testUserCredentials);
+
+    // Arrange: Define the incorrect old password and a new password
+    const incorrectOldPassword = 'incorrectSecret';
+    const newPassword = 'newSecret';
+
+    // Act: Attempt to update the user's password with an incorrect old password
+    const response = await request(app)
+      .patch('/api/v1/users/updateUserPassword')
+      .set('Cookie', [signedCookie])
+      .send({ oldPassword: incorrectOldPassword, newPassword });
+
+    // Assert: Check the response status for an error
+    expect(response.status).toBe(401);
+  });
+
+  it('should return an error if required data is missing', async () => {
+    // Arrange: Log in and get a signed cookie with valid test user credentials
+    const signedCookie = await loginAndReturnCookie(testUserCredentials);
+
+    // Act: Attempt to update the user's password without providing old and new passwords
+    const response = await request(app)
+      .patch('/api/v1/users/updateUserPassword')
+      .set('Cookie', [signedCookie])
+      .send({});
+
+    // Assert: Check the response status for an error
+    expect(response.status).toBe(400);
+  });
+});
+
+describe('DELETE /api/v1/users/deleteSingleUser endpoint', () => {
+  it('should delete a user if the user exists', async () => {
+    // Arrange: Log in and get a signed cookie with valid test user credentials
+    const signedCookie = await loginAndReturnCookie(testUserCredentials);
+
+    // Act: Attempt to delete the user using the authenticated cookie
+    const response = await request(app)
+      .delete('/api/v1/users/deleteSingleUser')
+      .set('Cookie', [signedCookie]);
+
+    // Assert: Check the response status for success
+    expect(response.status).toBe(200);
+    expect(response.body).toHaveProperty('msg', 'Success! User Deleted.');
+  });
```

Comment:
Currently, this test is failing because it's getting a 403 instead of a 200:

```

    Expected: 200
    Received: 403
```

We know that this endpoint is restricted to `admin` role users only. So the 403 makes sense because we're currently logging in as a user (Ava) whose role is just `user`. 

I would suggest creating an admin user (either within this test or in the `beforeEach` or `beforeAll` hook) and then logging in as that admin user for this success case test. 

The current test could still be useful as a separate test for verifying that we do indeed get a 403 when the logged in user is not an admin :)

Reply:
Two different tickets, for write and read endpoints. This is why two PR. But i probably going to merge this one with PS145. I need my admin from PS145 now))